Doc/library/tkinter.rst

```diff
@@ -11,100 +11,267 @@
-is maintained at ActiveState.)
+the Tcl/Tk GUI toolkit. Tcl/Tk and :mod:`tkinter` are available on most Unix
+platforms (including macOS), as well as on Windows systems.
 
 Running ``python -m tkinter`` from the command line should open a window
 demonstrating a simple Tk interface, letting you know that :mod:`tkinter` is
 properly installed on your system, and also showing what version of Tcl/Tk is
 installed, so you can read the Tcl/Tk documentation specific to that version.
 
-.. seealso::
-
-   Tkinter documentation:
-
-   `Python Tkinter Resources <https://wiki.python.org/moin/TkInter>`_
-      The Python Tkinter Topic Guide provides a great deal of information on using Tk
-      from Python and links to other sources of information on Tk.
-
-   `TKDocs <http://www.tkdocs.com/>`_
-      Extensive tutorial plus friendlier widget pages for some of the widgets.
-
-   `Tkinter reference: a GUI for Python <https://infohost.nmt.edu/tcc/help/pubs/tkinter/web/index.html>`_
-      On-line reference material.
-
-   `Tkinter docs from effbot <http://effbot.org/tkinterbook/>`_
-      Online reference for tkinter supported by effbot.org.
-
-   `Programming Python <http://learning-python.com/about-pp4e.html>`_
-      Book by Mark Lutz, has excellent coverage of Tkinter.
-
-   `Modern Tkinter for Busy Python Developers <https://www.amazon.com/Modern-Tkinter-Python-Developers-ebook/dp/B0071QDNLO/>`_
-      Book by Mark Rozerman about building attractive and modern graphical user interfaces with Python and Tkinter.
-
-   `Python and Tkinter Programming <https://www.manning.com/books/python-and-tkinter-programming>`_
-      Book by John Grayson (ISBN 1-884777-81-3).
-
-   Tcl/Tk documentation:
+Tkinter supports a range of Tcl/Tk versions, built either with or
+without thread support. The official Python binary release bundles Tcl/Tk 8.6
+threaded. See the source code for the :mode:`_tkinter` module
+for more information about supported versions.
+
+Tkinter is not a thin wrapper, but adds a fair amount of its own logic to
+make the experience more pythonic. This documentation will concentrate on these
+additions and changes, and refer to the official Tcl/Tk documentation for
+descriptions of underlying features.
+
+
+Architecture
+------------
+
+Tkinter provides a unified interface to several separate and distinct libraries.
+
+Tcl
+   Like Python, Tcl is a dynamic interpreted programming language. It is most
+   commonly used by C applications to embed a scripting language, or as an
+   interface to the Tk toolkit. The Tcl library has a C interface to
+   create and manage one or more instances of a Tcl interpreter, run Tcl
+   commands and scripts in those instances, and add custom commands
+   implemented in either Tcl or C. Tcl provides for an event queue, used
+   for I/O, timers, and by modules.
+   
+Tk
+   Tk is a module that can be loaded into a Tcl interpreter instance. It adds
```

Comment:
Agree. Thanks.

Question: Tcl uses the term "package" for what Python calls a module. Would it be better to say up front something to the effect of "the tcl interpreter can load packages to add features (written in C or Tcl, akin to Python modules), and then thereafter use "packages" when referring to Tcl add-ons? The alternative I think would be saying "Tcl module" everywhere vs. just "module" for Python.

Reply:
Use Tcl's official terminology to describe their relationship -- that way, the text will have an exact meaning.
No need to go into details: the reader only needs to know that: 1)it's some kind of loadable add-on; 2)it's implemented in C which (would explain how it can do GUI when Tcl itself can't); 3) it connects into Tcl by adding custom Tcl commands.

Reply:
https://github.com/roseman/cpython/pull/6